### ISSUE TITLE
Gnome3 .desktop file problem

### ISSUE THREAD

**Issue by z49x2vmq** [laurent22/joplin#2452]
After installing 1.0.179, Joplin is not shown in gnome's application list.

The Joplin icon is back when I remove version string, "1.0.179.940", from "appimagekit-joplin.desktop" file.

## Environment

Joplin version: 1.0.179
Platform: Linux
OS specifcs: Fedora 31

## Steps To Reproduce

1. Install or upgrade Joplin as guided in joplinapp.org

Describe what you expected to happen:
Joplin application menu item(Application Icon) is created in gnome's application list

**Comment by nezartarbin** [comment]
This may be related, but I built the same version from source. However, executing the `.AppImage` file does nothing.

Here is a relevant discussion:
https://discourse.joplinapp.org/t/appimage-from-current-master-not-running-on-linux/5667

**Comment by mic704b** [comment]
FYI that discussion is more about the current development version which is having a few build issues.

This discussion might be more relevant to 1.0.179:
https://discourse.joplinapp.org/t/unable-to-start-1-0-179-on-linux/5373

**Comment by bedwardly-down** [comment]
I don't know if this will solve your issue or not, but I created an unofficial patch that can't be currently merged into the main code due to it directly altering the AppImage and whatnot outside of being built from source. Just replace the version name with v1.0.179 in the instructions and you should be good to go.

https://github.com/laurent22/joplin/issues/2332#issuecomment-576909405

**Comment by domhnallw** [comment]
I've hit the same error as @z49x2vmq on my Ubuntu 18.04 x86-64 desktop. 

The `.desktop` file gets created, but it's got the version number inserted as the last line on its own rather than being a key-value pair:

```ini
[Desktop Entry]
Encoding=UTF-8
Name=Joplin
Comment=Joplin for Desktop
Exec=/home/<redacted>/.joplin/Joplin.AppImage
Icon=joplin
StartupWMClass=Joplin
Type=Application
Categories=Office;
1.0.179.940
``` 

If you try to use `update-desktop-database` to get Gnome to re-examine the file, it reports:

```sh
Could not parse file "/home/<redacted>/.local/share/applications/appimagekit-joplin.desktop": Key file contains line ?1.0.179.940? which is not a key-value pair, group, or comment
```

The culprit seems to be a reference to `$APPIMAGE_VERSION` [here](https://github.com/laurent22/joplin/blob/10cf80d6ca1dd6e96ba11f4ddd24c92b8c7a5bdc/Joplin_install_and_update.sh#L110), but I don't know enough about `.desktop` files to know if this is incomplete or what.

I know removing this last line in the file makes Gnome happy and that all of a sudden I get a desktop icon.

Hope this helps.

**Comment by harmlessdrudge** [comment]
Also with a fresh install on Linux Mint 19.3 using wget.
I got the icon by deleting the last line. 
Last line on a .175 version installed on another machine is
X-AppImage-BuildId=1UhCx8mcsln5JbH9RzSLD1tiHsy

**Comment by mrwulf** [comment]
Fixed by #2479
